Exception/UnknownComponentTypeException.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace Becklyn\GluggiBundle\Exception;
+
+
+/**
+ * Exception to indicate an unknown component type
+ */
+class UnknownComponentTypeException extends \InvalidArgumentException
+{
+    /**
+     * @param string          $type
+     * @param array           $knownTypes
+     * @param \Exception|null $previous
+     */
+    public function __construct (string $type, array $knownTypes = [], \Exception $previous = null)
+    {
+        $message = null !== $knownTypes
```

Comment:
This check seems redundant since you're enforcing an array with your type declaration.